maths/line_length.py

```diff
@@ -0,0 +1,62 @@
+from typing import Callable, Union
+import math as m
+
+def line_length(fnc: Callable[[Union[int, float]], Union[int, float]],
+                x_start: Union[int, float],
+                x_end: Union[int, float],
+                steps: int = 100) -> float:
+
+    """
+    Approximates the arc length of a line segment by treating the curve as a
+    sequence of linear lines and summing their lengths
+    :param fnc: a function which defines a curve
+    :param x_start: left end point to indicate the start of line segment
+    :param x_end: right end point to indicate end of line segment
+    :param steps: an accuracy gauge; more steps increases accuracy
+    :return: a float representing the length of the curve
+
+    >>> def f(x):
+    ...    return x
+    >>> '%.6f' % line_length(f, 0, 1, 10)
+    '1.414214'
+
+    >>> def f(x):
+    ...    return 1
+    >>> '%.6f' % line_length(f, -5.5, 4.5, 100)
```

Comment:
Please remove the trailing 100 parameter in this test to demonstrate that there is default value and it works.